lib/azure/armrest/model/storage_account.rb

```diff
@@ -119,6 +123,295 @@ def table_data(name, key = access_key, options = {})
         data
       end
 
+      ### Files and Directories
+
+      # Create a new directory under the specified +share+ or parent directory.
+      #
+      # The only supported option at this time is a "timeout" option.
+      #
+      def create_directory(share, directory, key = access_key, options = {})
+        raise ArgumentError, "No access key specified" unless key
+
+        query = "restype=directory"
```

Comment:
Can you have key and value pairs in a hash, and call `hash.to_query`? It improves readability.